store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -228,4 +243,118 @@ public CarbonReaderBuilder withRowRecordReader() {
     }
   }
 
+  /**
+   * gets an array of blocklet level CarbonInputSplits
+   * @return
+   * @throws IOException
+   */
+  public InputSplit[] getSplits() throws IOException {
```

Comment:
Instead of making as 2 methods, better make the single method and return an array of CarbonReaders and reuse the code which already existed.

Reply:
This is part of #3196, removed from here. will handle this comment there